week4/Program.cs

```diff
@@ -0,0 +1,156 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+var client = new HttpClient();
+
+//a get method that gets a response from an endpoint
+app.MapGet("/", async () =>
+{
+    var result = await client.GetFromJsonAsync<UsersData>("https://reqres.in/api/users?page=2");
+    return result;
+});
+
+//a post method that posts to an endpoint e.g. /api/users
+app.MapPost("/", async () =>
+{
+    var result = await client.PostAsJsonAsync("https://reqres.in/api/users", new User(18, 31, "D", "J", "G"));
+    return Results.Ok(result.IsSuccessStatusCode);
+});
+
+//A POST endpoint that calls POST users/add with a record with FirstName, LastName and Age
+app.MapPost("/users/add", async (User user) =>
+{
+    var response = await client.PostAsJsonAsync("https://dummyjson.com/users/add", user);
+    var createdUser = await response.Content.ReadFromJsonAsync<User>();
+    var result = new
+    {
+        StatusCode = response.StatusCode,
+        data = createdUser,
+    };
+    return Results.Ok(result);
+});
+
+//A POST endpoint that that calls Post products/add with a record with Title and Price (this simulates creating a product)
+app.MapPost("/products/add", async (Product product) =>
+{
+    var response = await client.PostAsJsonAsync("https://dummyjson.com/products/add", product);
+    var createdProduct = await response.Content.ReadFromJsonAsync<Product>();
+    var result = new
+    {
+        StatusCode = response.StatusCode,
+        data = createdProduct,
+    };
+    return Results.Ok(result);
+});
+
+
+//A POST endpoint that takes a lists of ids and retrieves all of the users with those ids from the GET users (Id, FirstName, LastName and Age)
+app.MapPost("/users/list", async (IdArray Array) =>
```

Comment:
try to avoid using keywords as parameter/variable names, in this case `Array`. This can be `userIds` for example